### ISSUE TITLE
socket might not exist in callback of setImmediate

### ISSUE THREAD

**Issue by kapouer** [edasarl/socket.io-backlog#2]
```
2015-03-23_20:30:31.96745 TypeError: Cannot call method 'on' of undefined
2015-03-23_20:30:31.96746     at Object._onImmediate (/home/eda/public_nodejs/nsocket-core/production/node_modules/socket.io-backlog/index.js:121:12)
2015-03-23_20:30:31.96746     at processImmediate [as _immediateCallback] (timers.js:336:15)
```

**Comment by manubb** [comment]
Is this easy to reproduce?

**Comment by kapouer** [comment]
not that easy... maybe when someone leaves early

**Comment by kapouer** [comment]
https://github.com/Automattic/socket.io/pull/2063

**Comment by manubb** [comment]
This should be fixed in version 0.0.3.
